client/mainmenu/CCampaignScreen.cpp

```diff
@@ -101,11 +102,24 @@ CCampaignScreen::CCampaignButton::CCampaignButton(const JsonNode & config)
 	campFile = config["file"].String();
 	video = VideoPath::fromJson(config["video"]);
 
-	status = config["open"].Bool() ? CCampaignScreen::ENABLED : CCampaignScreen::DISABLED;
+	status = CCampaignScreen::ENABLED;
 
 	auto header = CampaignHandler::getHeader(campFile);
 	hoverText = header->getName();
 
+	if(persistent["campaign"][campaignSet][header->getFilename()]["completed"].Bool())
+		status = CCampaignScreen::COMPLETED;
+
+	for(const JsonNode & node : parentConfig[campaignSet]["items"].Vector())
+	{
+		for(const JsonNode & requirement : config["requires"].Vector())
+		{
+			if(node["id"].Integer() == requirement.Integer())
+				if(!persistent["campaign"][campaignSet][node["file"].String()]["completed"].Bool())
+					status = CCampaignScreen::DISABLED;
```

Comment:
Perhaps add some option to override this and have all campaigns unlocked without completion? E.g. `persistent["unlockAllCampaigns"].Bool()`